api/datastore/sql/sql_test.go

```diff
@@ -40,6 +41,18 @@ func newWithMigrations(url *url.URL) (*sqlStore, error) {
 	return ds, nil
 }
 
+func retryDBConnection(t *testing.T, attempts int, sleep time.Duration, dbURL *url.URL) (err error) {
+	for i := 0; i < attempts; i++ {
+		t.Logf("[%v] - attempting to reach out datastore %v", i, dbURL.String())
+		_, err = newDS(dbURL)
```

Comment:
this generates a lot of new db instances, which isn't really cheap afaik and also doesn't really ensure that the datastore instance we're going to actually use is valid and adds extra code in the db initialization caller path to call retryDBConnection. is there any reason not to do just do this in our db initialization `ping` inside of the sql package?

Reply:
fixed, see #630, I'll rebase this PR to that branch.